Assets/PatchKit Patcher/Scripts/Patcher.cs

```diff
@@ -916,6 +927,80 @@ private void ThreadUpdateApp(bool automatically, CancellationToken cancellationT
                 }
             }
         }
+        
+        private void ThreadVerifyAllAppFiles(CancellationToken cancellationToken)
+        {
+            _state.Value = PatcherState.UpdatingApp;
+
+            _updateAppCancellationTokenSource = new PatchKit.Unity.Patcher.Cancellation.CancellationTokenSource();
+   
+            using (cancellationToken.Register(() => _updateAppCancellationTokenSource.Cancel()))
+            {
+                _isAppInstalled.Value = false;
+                
+                var appUpdater = new AppUpdater.AppUpdater(new AppUpdaterContext(_app, _configuration.AppUpdaterConfiguration));
+
+                try
+                {
+                    _updaterStatus.Value = appUpdater.Status;
+
+                    using (_updaterStatus.Take(1).Subscribe((status) => _state.Value = PatcherState.UpdatingApp))
+                    {
+                        appUpdater.VerifyFiles(cancellationToken);
+                    }
+                }
+                catch (OperationCanceledException)
+                {
+                    PatcherStatistics.DispatchSendEvent(PatcherStatistics.Event.PatcherCanceled);
+
+                    throw;
+                }
+                finally
+                {
+                    _state.Value = PatcherState.None;
+
+                    _updaterStatus.Value = null;
+                    _updateAppCancellationTokenSource = null;
+                }
+            }
+        }
+        
+        private void ThreadUninstallApp(CancellationToken cancellationToken)
+        {
+            _state.Value = PatcherState.UpdatingApp;
```

Comment:
Please introduce here a `// TODO:` comment as we might want to introduce here a new state.

Reply:
done